test/functional/wallet_tokens_freeze.py

```diff
@@ -147,6 +148,11 @@ async def async_test(self):
 
             assert_in("The transaction was submitted successfully", await wallet.freeze_token(token_id, 'unfreezable'))
 
+            # randomly put the tx in a block or keep in mempool unconfirmed
+            if random.choice([True, False]):
```

Comment:
As you see in the tests in the node, we use a fixed seed to reproduce the exact same error. While I don't mind in this case because the choices are True/False, usually I'd print that outcome in the logs so that I can see what happened instead of having to guess.

Same applies to the other random.choice() below.

Also... Does the comment make sense?

Reply:
changed the seed log to info so we can see when a test fails what was the random seed number.

The comment means if true we will produce a new block and the wallet will sync and have the previous transaction as confirmed, otherwise it will be unconfirmed.
There is different logic in the wallet as it is fetching the token state from the chainstate which only contains the confirmed status, and for unconfirmed txs we apply them on top of that state in the wallet.